packaging/public-api/build.gradle.kts

```diff
@@ -28,19 +31,50 @@ dependencies {
 
     // Groovy is part of our API
     externalApi(libs.groovy)
+    // Groovy modules that are part of our API
+    // See DependencyClassPathProvider and DependencyClassPathNotationConverter
+    externalApi(libs.groovyAnt)
```

Comment:
❓ Since we are expanding on dependencies, maybe consider [this comment](https://github.com/gradle/gradle/issues/29483#issuecomment-3078805705)

Reply:
As discussed with Louis, I will restore the `-internal` variant as it was and introduce a new one, not-published, with the complete list of dependencies that the current generated API jar has in order to limit the potential for this PR breaking existing builds.

Reply:
Done